### ISSUE TITLE
azure - metric for number of API calls

### ISSUE THREAD

**Issue by aluong** [cloud-custodian/cloud-custodian#4826]
Analogously with AWS, we should be able to record the number of API calls as a metric. We need to hook into the [Azure SDK pipeline](https://github.com/Azure/msrest-for-python/wiki/msrest-0.6.0---Pipeline) to record this information.

**Comment by stefangordon** [comment]
Yeah I vaguely remember we have a hook there already in session.py to configure some retry stuff.
